### PR TITLE
Change ATEN generator argument type to const std::optional<Generator>&

### PR DESCRIPTION
**Pull request by cyyever**
To adopt the changes in https://github.com/pytorch/pytorch/pull/120076

### CODE REVIEW

**Comment by cyyever** [comment]
@alanwaketan Please help review it

File: torch_xla/csrc/aten_xla_type.cpp [hunk 1]
```diff
@@ -59,6 +59,12 @@
 //   https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/templates/Operators.h
 //   (You can find some examples below)
 
+#if defined(USE_TORCH_NEW_OPTIONAL_GENERATOR_TYPE)
```
Comment (alanwaketan) [inline torch_xla/csrc/aten_xla_type.cpp:62]: Any reasons why we use a marco here?

Comment (alanwaketan) [reply torch_xla/csrc/aten_xla_type.cpp:62]: I don't think we need this guard. You can add a .torch_pin to this PR which will then pick up your PR on the upstream. So we can land those together!
https://github.com/pytorch/xla/tree/db64a45469acb20d6b75299b6434f9e11c1c95c2/torch_patches

Comment (cyyever) [reply torch_xla/csrc/aten_xla_type.cpp]: Fixed

**Comment by alanwaketan** [comment]
Great, looks like all tests are passed. Please ping me again once the upstream PR is approved. Also, we need to remove the .torch_pin before landing the PR.

**Comment by cyyever** [comment]
> Great, looks like all tests are passed. Please ping me again once the upstream PR is approved. Also, we need to remove the .torch_pin before landing the PR.

The upstream PR has been approved. Should we first merge the upstream PR or this PR?

**Comment by alanwaketan** [comment]
> > Great, looks like all tests are passed. Please ping me again once the upstream PR is approved. Also, we need to remove the .torch_pin before landing the PR.
> 
> The upstream PR has been approved. Should we first merge the upstream PR or this PR?

The upstream PR.

**Review by alanwaketan — APPROVED** [review]
LGTM. Thanks for making the companion change!

**Comment by cyyever** [comment]
@alanwaketan It seems that I have to merge this PR first, otherwise, how can the upstream PR have a valid commit hash to be filled in .github/ci_commit_pins/xla.txt? Or what you mean was that I can just merge by force while ignoring the upstream XLA test failure?

**Comment by cyyever** [comment]
@alanwaketan I have a confusion. If this PR is not merged first, I can't get a valid commit hash in XLA to be referred in the upstream PR for passing its XLA test.

**Comment by alanwaketan** [comment]
Oh, I forgot that xla hash doesn't work for forked pull request. Do you mind if I took the PR and make a new PR through branching?

**Comment by cyyever** [comment]
> Oh, I forgot that xla hash doesn't work for forked pull request. Do you mind if I took the PR and make a new PR through branching?

 Please help me.

**Comment by alanwaketan** [comment]
Close this in favor of https://github.com/pytorch/xla/pull/6686

**Pull request closed without merging**